### PR TITLE
Tarball compression

### PR DESCRIPTION
**Pull request by kirillgarbar**
Added compression of uploaded and downloaded tarballs using zlib.
Compression stage can be added to other pipelines if needed.

### CODE REVIEW

File: ch_backup/compression/zlib.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    compressobj,
+    decompressobj,
+)
+
+from ch_backup.compression.base import BaseCompression
+
+Z_DEFAULT_WBITS = 15
+
+
+class ZLIBCompression(BaseCompression):
```
[the remark this answers is not in the document]
Comment (aalexfvk) [reply ch_backup/compression/zlib.py:18]: And, please, add unit test for this class

**Review by aalexfvk — CHANGES_REQUESTED** [review]

**Comment by aalexfvk** [comment]
Try to add type hints for all new code. It is not forced by linters yet, but will help in the future to enable this checking.

File: ch_backup/storage/async_pipeline/stages/compression/compress_stage.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    Compresses data and return it by chunks of specified size
+    """
+
+    stype = StageType.COMPRESSION
+
+    def __init__(
+        self, compressor: BaseCompression, chunk_size: int, buffer_size: int
+    ) -> None:
+        self._compressor = compressor
+        self._buffer = BytesFIFO(buffer_size)
```
[the remark this answers is not in the document]
Comment (kirillgarbar) [reply ch_backup/storage/async_pipeline/stages/compression/compress_stage.py:24]: This also means we should not aggregate data until `chunk_size` is reached?

[the remark this answers is not in the document]
Comment (aalexfvk) [reply ch_backup/storage/async_pipeline/stages/compression/compress_stage.py:24]: Yes next `ChunkingStage` should do that. This is its responsibility.

File: ch_backup/backup/metadata/cloud_storage_metadata.py [hunk 3]
```diff
@@ -54,13 +65,18 @@ def load(cls, data: Dict[str, Any]) -> "CloudStorageMetadata":
         """
         Deserialize Cloud Storage metadata.
         """
-        return cls(encryption=data.get("encryption", True), disks=data.get("disks", []))
+        return cls(
+            encryption=data.get("encryption", True),
+            compression=data.get("compression", True),
```
[the remark this answers is not in the document]
Comment (aalexfvk) [reply ch_backup/backup/metadata/cloud_storage_metadata.py]: Almost done. I meant that we should treat absent of `compression` key in the metadata as `False`, e.g. for existing backups, but use compression for new ones. Now you can just set this default in config.

**Review by aalexfvk — APPROVED** [review]

**Pull request merged**